### ISSUE TITLE
Disconnect peers randomly to have a better peer diversity

### ISSUE THREAD

**Issue by MithrilMan** [stratisproject/StratisBitcoinFullNode#2827]
Currently the fullnode stay always connected to the same peer if they are "theoretically" behaving good.
Once the node establish a connection (both inbound or outbound) it stays connected to the peer until that peer disconnect or break some rule that causes that peer to be banned.

Problem:
Imagine that those nodes sync with us properly, announce blocks, etc.. but they don't relay our transactions, our node could be isolated.
This potential problem is even more probable for peers that are behind a Firewall and maybe can't have peers connected to them (so 0 inbound connections)

Solution:
What I propose is a way to increase peer diversity and reduce possible attacks of this nature.
Basically everytime we connect to a peer (so for outbound connections) we set a property (let's call it `ConnectionExpiration`) whose value is set between a minimum and a maximum timespan value.
Then we have a task that runs every minute that checks which peer is expired, and disconnect it from the node, banning it for a specified duration (to prevent we reconnect to it too soon)

**Comment by zeptin** [comment]
Sounds good, but my first question would be whether Core do something similar or not. If they do not, there is probably a good reason.

**Comment by MithrilMan** [comment]
Or they didn't thought about that :D (joking, need to investigate)
I forget about saying that whitelisted peers should never been affected by this (I'll edit my problem description).
